app/uk/gov/hmrc/helptosavereminder/controllers/EmailCallbackController.scala

```diff
@@ -17,49 +17,45 @@
 package uk.gov.hmrc.helptosavereminder.controllers
 
 import com.google.inject.Inject
-
 import play.api.Logger
 import play.api.libs.json.{JsError, JsSuccess, Json}
 import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
-
 import uk.gov.hmrc.helptosavereminder.audit.HTSAuditor
 import uk.gov.hmrc.helptosavereminder.config.AppConfig
 import uk.gov.hmrc.helptosavereminder.models.{EventsMap, HtsReminderUserDeleted, HtsReminderUserDeletedEvent}
 import uk.gov.hmrc.helptosavereminder.repo.HtsReminderMongoRepository
 import uk.gov.hmrc.http.HttpClient
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
-
 import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
 import uk.gov.hmrc.helptosavereminder.util.JsErrorOps._
-
 import cats.instances.string._
 import cats.syntax.eq._
+import uk.gov.hmrc.helptosavereminder.connectors.EmailConnector
 
 import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success}
 
 class EmailCallbackController @Inject()(
   http: HttpClient,
   servicesConfig: ServicesConfig,
   val cc: MessagesControllerComponents,
   repository: HtsReminderMongoRepository,
-  auditor: HTSAuditor)(implicit ec: ExecutionContext, appConfig: AppConfig)
+  auditor: HTSAuditor,
+  emailConnector: EmailConnector)(implicit ec: ExecutionContext, appConfig: AppConfig)
     extends BackendController(cc) {
 
   def handleCallBack(callBackReference: String): Action[AnyContent] = Action.async { implicit request =>
     request.body.asJson.map(_.validate[EventsMap]) match {
       case Some(JsSuccess(eventsMap, _)) ⇒ {
         if (eventsMap.events.exists(x => (x.event === "PermanentBounce"))) {
-          val nino = callBackReference.takeRight(9)
-          Logger.info(s"Reminder Callback service called for NINO = $nino")
-          repository.findByNino(nino).flatMap {
+          Logger.info(s"Reminder Callback service called for callBackReference = $callBackReference")
+          repository.findByCallBackUrlRef(callBackReference).flatMap {
             case Some(htsUser) =>
               val url = s"${servicesConfig.baseUrl("email")}/hmrc/bounces/${htsUser.email}"
               Logger.debug(s"The URL to request email deletion is $url")
-              repository.deleteHtsUserByCallBack(nino, callBackReference).flatMap {
+              repository.deleteHtsUserByCallBack(htsUser.nino.nino, callBackReference).flatMap {
```

Comment:
Please use nino.value in place of nine.nine throughout

Reply:
Done